### ISSUE TITLE
This action fails in combination with Python 3.10

### ISSUE THREAD

**Issue by csadorf** [s-weigand/setup-conda#169]
**Describe the bug**

Unable to use this step in combination with Python version "3.10".

The bug occurred as part of a [workflow](https://github.com/aiidateam/aiida-core/runs/4211991423?check_suite_focus=true) defined [here](https://github.com/aiidateam/aiida-core/blob/5532a8e0527b75f52addcfc164fb53a4ba0f73de/.github/workflows/test-install.yml#L76-L92).

**To Reproduce**

The following workflow definition should reproduce the problem:

```yaml
  create-conda-environment:
    runs-on: ubuntu-latest

    - name: Setup Conda
      uses: s-weigand/setup-conda@v1
      with:
        python-version: '3.10'
        update-conda: false
        conda-channels: conda-forge
```

**Expected behavior**

The action creates a conda environment with Python 3.10.

**Comment by s-weigand** [comment]
@csadorf Thanks for reporting this bug! 
Looks like this is due conflicts with the packages preinstalled in the conda base env on the runner.
I guess if a `python-version` is specified creating a new env and activating this should be faster anyway (hope I find time at the weekend to fix this).

That said I also fiddled around with the test you try to run and it looks like it doesn't do what you expect.
My guess of what you want to do is, that you want to verify that you can install the env in `environment.yml` on python 3.10.

A workaround for the previous behavior would be to install a python 3.10 env and run the code from there
E.g. with [this workflow](https://github.com/s-weigand/setup-conda/actions/runs/1464085810/workflow) and the env resolution test passes. But when you have a look at commands output you see that it used python 3.9 (`- conda-forge/linux-64::python==3.9.7=hb7a2778_3_cpython`)

So if you want to test that the dependencies can be installed on python 3.10 you need to replace `python~=3.7` in the  `environment.yml` file with `python~=3.10`
https://github.com/s-weigand/setup-conda/blob/bbb1fe76acdd4abe10d8e0b8948760140c560a24/.github/workflows/reproduce-issue-169.yml#L1-L40
And in that case, [it fails](https://github.com/s-weigand/setup-conda/runs/4217254803?check_suite_focus=true) due to the too strictly pinned versions.

Hope these workarounds help until I got time to fix the issue.

**Comment by s-weigand** [comment]
@all-contributors please add @csadorf  for bug

**Comment by csadorf** [comment]
@s-weigand Thank you very much for the support and also for reviewing our workflow and even proposing an improvement. 👍 

Indeed, the specification of a specific Python version as part of [the job you replicated](https://github.com/aiidateam/aiida-core/blob/01a88469e820e43a5a0e749b49476dfa025c4ec0/.github/workflows/test-install.yml#L76-L100) has no effect and I would only want to keep it to avoid unexpected failures related to the action.

However, the actual tests for checking whether the package can be installed for a specific Python version is implemented [here](https://github.com/aiidateam/aiida-core/blob/01a88469e820e43a5a0e749b49476dfa025c4ec0/.github/workflows/test-install.yml#L152-L164):
```yaml
    - name: Setup Conda
      uses: s-weigand/setup-conda@v1
      with:
        python-version: ${{ matrix.python-version }}
        update-conda: false
        conda-channels: conda-forge
    - run: conda --version
    - run: python --version
    - run: which python


    - name: Test direct installation
      run: |
        conda create --dry-run -n test-install aiida-core python=${{ matrix.python-version }}
```
And unless I am missing something it should work as expected.

> Hope these workarounds help until I got time to fix the issue.

I will implement the workaround for now, so that we can move forward. Thank you!

**Comment by s-weigand** [comment]
This should be fixed in the [v1.1.0 release](https://github.com/s-weigand/setup-conda/releases/tag/v1.1.0).
